test/integ/basic.cpp

```diff
@@ -1021,6 +1021,33 @@ LT_BEGIN_AUTO_TEST(basic_suite, url_with_regex_like_pieces)
     curl_easy_cleanup(curl);
 LT_END_AUTO_TEST(url_with_regex_like_pieces)
 
+LT_BEGIN_AUTO_TEST(basic_suite, method_not_allowed_header)
+    simple_resource resource;
+    resource.disallow_all();
+    resource.set_allowing("POST", true);
+    resource.set_allowing("HEAD", true);
+    ws->register_resource("base", &resource);
+    curl_global_init(CURL_GLOBAL_ALL);
+    string s;
+    map<string, string> ss;
+    CURL *curl = curl_easy_init();
+    CURLcode res;
+    curl_easy_setopt(curl, CURLOPT_URL, "localhost:8080/base");
+    curl_easy_setopt(curl, CURLOPT_HTTPGET, 1L);
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writefunc);
+    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &s);
+    curl_easy_setopt(curl, CURLOPT_HEADERFUNCTION, headerfunc);
+    curl_easy_setopt(curl, CURLOPT_HEADERDATA, &ss);
+    res = curl_easy_perform(curl);
+    LT_ASSERT_EQ(res, 0);
+    long http_code = 0;
+    curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
```

Comment:
cpplint complains about this, however according to https://curl.se/libcurl/c/CURLINFO_RESPONSE_CODE.html *long* is the correct type for `curl_easy_getinfo()` here.

Reply:
Used `int64_t` like elsewhere in that file now. 🤷🏼‍♂️